__tests__/helpers/setup-oclif-config.ts

```diff
@@ -0,0 +1,42 @@
+import type { Command as OclifCommand } from '@oclif/core';
+
+import path from 'node:path';
+
+import { Config } from '@oclif/core';
+import { captureOutput } from '@oclif/test';
+
+/**
+ * Used for setting up the oclif configuration for simulating commands in tests.
+ * This is a really barebones approach so we can continue using vitest + nock
+ * how we want to.
+ *
+ * @see {@link https://github.com/oclif/test}
+ * @see {@link https://oclif.io/docs/testing}
+ */
+export default function setupOclifConfig() {
+  // https://stackoverflow.com/a/61829368
+  const root = path.join(new URL('.', import.meta.url).pathname, '.');
+
+  return Config.load({
+    root,
+    version: '7.0.0',
+  });
+}
+
+export function runCommand<T extends typeof OclifCommand>(Command: T) {
+  return async function runCommandAgainstArgs(args?: string[]) {
+    const oclifConfig = await setupOclifConfig();
+    // @ts-expect-error this is the pattern recommended by the @oclif/test docs.
+    // Not sure how to get this working with type generics.
```

Comment:
Looked into this a bit and I don't think it's possible for us to do because arguments and flags are supplied differently to the CLI and need actual string parsing to determine what you're doing.

Reply:
yeah so the goal here was to make it so `npx vitest --watch [command-file].ts` properly reloads every time i make a change to the command file (which is surprisingly challenging b/c `oclif` does a lot of magical command module importing by default). this TS error isn't great but otherwise i'm pretty content with this setup overall (open to ideas though)